src/tun.c

```diff
@@ -530,9 +544,12 @@ read_tun(int tun_fd, char *buf, size_t len)
 static int
 tun_uses_header(void)
 {
-#if defined (FREEBSD) || defined (NETBSD)
-	/* FreeBSD/NetBSD has no header */
-	return 0;
+#if defined FREEBSD || defined NETBSD || defined OPENBSD
+        /* To enable IPv6 in FreeBSD tunnels, tunnel
+         * headers now enabled for that platform
+         */
+        return 1;
```

Comment:
This requires a bump of protocol version (src/version.h) - is not backwards compatible

Also OpenBSD mentioned in the #else case below